### ISSUE TITLE
Feature Request: extending trellis check command

### ISSUE THREAD

**Issue by sangemaru** [roots/trellis-cli#376]
### Terms

- [X] I have read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
- [X] This request is not a duplicate of an existing issue
- [X] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

### Summary

Extend the dependency checks the `trellis check` command carries out to include required provisioning/deployment checks like SSH agent, keys, Ansible.

#### Why are we doing this?
Right now, `trellis check` looks for Python, Vagrant and Virtualbox compatibility related to local development, but some important checks would be:
- system Ansible version and its compatibility
- SSH agent being enabled
- SSH agent keys being loaded

#### What use cases does it support?
Would be nice to support checks of any known dependencies of the deploy/provision processes.

#### What is the expected outcome?
An output like:
```
Checking Trellis requirements...

Required:
[✓] Ansible [>= 2.9.7]: 2.10.0
[✓] Python [>= 3.8.0]: 3.11.2
[✓] SSH Agent [On]: Yes
[✓] SSH Agent [Loaded]: Yes

Optional:

[✓] Vagrant [>= 2.1.0]: 2.3.4
[✓] VirtualBox [>= 4.3.10]: 6.1.36r152435
...
```
#### Potential conflicts / foreseeable issues
May conflict with Ansible version in virtual environment

**Comment by swalkinshaw** [comment]
`check` definitely needs some improvements and I like some of these ideas.

That command was also added years before the CLI could create a virtualenv for better dependency handling. It could say if you're run `trellis init`?

```plain
Checking Trellis requirements...

[✓] Project initialized

Required:
[✓] Ansible [>= 2.9.7]: 2.10.0
[✓] Python [>= 3.8.0]: 3.11.2
[✓] SSH Agent [On]: Yes
[✓] SSH Agent [Loaded]: Yes

Optional:

[✓] Vagrant [>= 2.1.0]: 2.3.4
[✓] VirtualBox [>= 4.3.10]: 6.1.36r152435
...

```

I'll give this some more thought.

**Comment by sangemaru** [comment]
Yeah, I use `trellis init`. But it doesn't check for those basic requirements, so we need to remember to do those manually. On my end, that's rarely an issue, but I usually can't hand off trellis-related processes to others because of all the details they're not aware of. So either sticking it into `trellis check` or `trellis init` works
